packages/wonder-blocks-dropdown/components/multi-select.js

```diff
@@ -136,11 +147,25 @@ export default class MultiSelect extends React.Component<Props, State> {
         };
     }
 
-    handleOpenChanged = (open: boolean, keyboard?: boolean) => {
+    /**
+     * Used to sync the `opened` state when this component acts as a controlled
+     * component
+     */
+    static getDerivedStateFromProps(props: Props, state: State) {
+        return {
+            open: typeof props.opened === "boolean" ? props.opened : state.open,
+        };
+    }
+
+    handleOpenChanged = (opened: boolean, keyboard?: boolean) => {
         this.setState({
-            open,
+            open: opened,
             keyboard,
         });
+
+        if (this.props.onToggle) {
+            this.props.onToggle(opened);
+        }
     };
 
     handleToggle = (selectedValue: string) => {
```

Comment:
~It might be a little confusing to someone looking at this latter why we don't call `this.props.onToggle`.  Can you add some comments here.  Maybe it would be good to name the prop `onToggleMenu` and then this event handler could be called `handleToggleSelection`.~

Reply:
I think a more drastic refactoring though might be in order in a future PR.  I noticed that action-menu.js uses `Dropdown` but here we use `DropdownCore` directly.  I think we should be able to use `Dropdown` here since it has a `selectionType` prop now.

Reply:
This work will be also included in this ticket: https://khanacademy.atlassian.net/browse/WB-634